File: MineInAbyss/src/main/java/com/derongan/minecraft/mineinabyss/Ascension/Effect/Effects/DeathAscensionEffect.java
package com.derongan.minecraft.mineinabyss.Ascension.Effect.Effects;

import org.bukkit.entity.Player;

public class DeathAscensionEffect extends AbstractAscensionEffect {
    public DeathAscensionEffect(int offset, int strength, int duration, int iterations) {
        super(offset, strength, duration, iterations);
    }

    @Override
    void applyEffect(Player player) {
        player.setHealth(0.0D);
    }
}

File: mineinabyss-systems/src/main/kotlin/com/mineinabyss/curse/effects/DeathAscensionEffect.kt
@file:UseSerializers(DurationSerializer::class)

package com.mineinabyss.curse.effects

import com.mineinabyss.idofront.serialization.DurationSerializer
import com.mineinabyss.idofront.time.ticks
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import org.bukkit.entity.Player
import kotlin.time.Duration

@Serializable
@SerialName("death")
data class DeathAscensionEffect(
    override val offset: Duration = 0.ticks,
    override val duration: Duration = 0.ticks,
    override val iterations: Int = 0
) : AbstractAscensionEffect() {
    override fun applyEffect(player: Player) {
        player.health = 0.0
    }

    override fun clone() = copy()
}
